tests/php/ORM/ExactMatchFilterTest.php

```diff
@@ -0,0 +1,106 @@
+<?php
+
+namespace SilverStripe\ORM\Tests;
+
+use SilverStripe\Core\Config\Config;
+use SilverStripe\Dev\SapphireTest;
+use SilverStripe\ORM\Filters\ExactMatchFilter;
+use SilverStripe\ORM\Tests\ExactMatchFilterTest\Task;
+use SilverStripe\ORM\Tests\ExactMatchFilterTest\Project;
+
+class ExactMatchFilterTest extends SapphireTest
+{
+    protected $usesDatabase = true;
+
+    protected static $extra_dataobjects = [
+        Task::class,
+        Project::class,
+    ];
+
+    protected function setUp(): void
+    {
+        parent::setUp();
+        for ($i = 0; $i < 3; $i++) {
+            Project::create([
+                'Title' => "Project $i"
+            ])->write();
+            Task::create([
+                'Title' => "Task $i",
+                'ProjectID' => $i,
+            ])->write();
+        }
+    }
+
+    /**
+     * @dataProvider provideUsePlaceholders
+     */
+    public function testUsePlaceholders(?bool $expectedID, ?bool $expectedTitle, $config, callable $fn): void
+    {
+        Config::modify()->set(ExactMatchFilter::class, 'use_placeholders_for_integer_ids', $config);
+        [$idQueryUsesPlaceholders, $titleQueryUsesPlaceholders] = $this->usesPlaceholders($fn);
+        $this->assertSame($expectedID, $idQueryUsesPlaceholders);
+        $this->assertSame($expectedTitle, $titleQueryUsesPlaceholders);
+    }
+
+    public function provideUsePlaceholders(): array
+    {
+        $ids = [1, 2, 3];
+        $taskTitles = array_map(fn($i) => "Task $i", $ids);
+        return [
+            'primary key' => [
+                false,
+                null,
+                false,
+                fn() => Task::get()->byIDs($ids)->toArray()
```

Comment:
Please use the parameter names as keys, e.g:
```suggestion
                'expectedID' => false,
                'expectedTitle' => null,
                'config' => false,
                'fn' => fn() => Task::get()->byIDs($ids)->toArray()
```
This makes it a lot easier to see at a glance what we're using these values for.

Reply:
Updated